tests/kafkatest/services/kafka/kafka.py

```diff
@@ -292,10 +292,18 @@ def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAI
                 use_new_coordinator = context.injected_args.get(arg_name)
             if use_new_coordinator is None:
                 use_new_coordinator = context.globals.get(arg_name)
+
+        # Set use_share_groups based on injected arguments.
```

Comment:
Why is the handling of `use_share_groups` different than `use_transactions_v2`? It seems to me that this is different from both `use_new_coordinator` and `use_transactions_v2`, and I cannot see why it shouldn't just follow the simpler `use_transactions_v2` pattern.